packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -272,6 +272,39 @@ export class EntityService {
     }
   }
 
+  private async checkServiceEntityLicense(resource: Resource) {
+    if (!this.billingService.isBillingEnabled) {
+      return;
+    }
+    await this.checkServiceLicense(resource);
+
+    const serviceEntityEntitlementPromise =
+      this.billingService.getNumericEntitlement(
+        resource.project.workspaceId,
+        BillingFeature.EntitiesPerService
+      );
+
+    const resourceEntitiesCount = this.prisma.entity.count({
+      where: {
+        resourceId: resource.id,
+        deletedAt: null,
+      },
+    });
+
+    const serviceEntityEntitlement = await serviceEntityEntitlementPromise;
```

Comment:
@overbit  Why not do the await in the functions calls?

Reply:
has prisma and http call are not dependent on each other, we are going to await only when we need their results

Reply:
This const : serviceEntityEntitlementPromise not related to prisma, 

I mean: 
  const serviceEntityEntitlementPromise = **await** 
      this.billingService.getNumericEntitlement(
        resource.project.workspaceId,
        BillingFeature.EntitiesPerService
      );

Am I missing something??

Reply:
The code is firing the following 2 promises: `this.billingService.getNumericEntitlement` and `this.prisma.entity.count`.
As they are not dependent on each other, we can have the await later where is needed to avoid blocking the async execution on one of these statements